### DISCUSSION TITLE
Runtime load and return GameObject, not just a bool

### DISCUSSION THREAD

**Discussion by andypoly** [atteneder/glTFast#404] [Q&A]
Hi I realise I can create a custom IInstantiator, but I wondered why there was just the runtime "bool InstantiateMainScene( Transform parent )" call and not one that returned a GameObject in root of scene which you can then place somewhere or hide and use/duplicate later?
If you want to refer to the loaded GameObject later you have to find it under the parent Transform with that call.
Am I missing something?

**Comment by atteneder** [comment]
First off, the idea was that users who need it create their own root GameObjects upfront and pass it as `parent`. Re-parenting, transforming and (de-)activating can be done on said GameObject.

But I agree that you can make this argument, since explicit is better than implicit. Draft:

```csharp
public GameObject InstantiateMainScene( Transform parent )
```

But what should it return exactly? The scene GameObject seems to be the logical answer for the status quo, but there are plans to make that one optional (see #320). The options for creating a scene GameObject could be:

1. Always
2. Only if there are more than one root nodes
3. Never (not available for design-time import, since Prefabs need a root transform)

For Case three returning a single GameObject (or Transform) is not sufficient. You could return the parent, but that's confusing imho.

Happy to get you thoughts about it.

P.S.: Removing the status quo is out of question, since that would be a breaking change.

**Reply by andypoly** [reply]
> First off, the idea was that users who need it create their own root GameObjects upfront and pass it as parent

But the scene object is currently a parent already so no need for another one - just adds to Transform stack.

I was going to add `public GameObject InstantiateMainScene() ` to create the model with its scene node (returning that) and put in scene, then I can move it, hide it, instantiate multiple times whatever.
My preference might be to remove the scene node where there is one node below but not if multiple children

If you are going to remove the scene node and there's a chance of needing a parent node to encompass all the imported models, then it is not for that

**Reply by atteneder** [reply]
I think #320 has to be resolved first (just decided, not necessarily implemented) before fixing this (to avoid breaking API shortly after).

Thanks
